src/Sulu/Bundle/AdminBundle/Resources/js/components/Modal/Modal.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+import React from 'react';
+import {observable, action} from 'mobx';
+import {observer} from 'mobx-react';
+import Portal from 'react-portal';
+import classnames from 'classnames';
+import {afterElementsRendered} from '../../services/DOM';
+import Backdrop from '../Backdrop';
+import ModalBox from './ModalBox';
+import type {ModalProps} from './types';
+import modalStyles from './modal.scss';
+
+type Props = ModalProps & {
+    isOpen: boolean,
+    onRequestClose: () => void,
+};
+
+const ESC_KEY = 27;
```

Comment:
`react-keydown` only listens to keydowns within the component elements. Because we use portals in the modal it doesn't really work. I created an issue in `react-keydown`: https://github.com/glortho/react-keydown/issues/68